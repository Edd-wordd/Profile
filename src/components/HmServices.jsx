import React from 'react'
import mainTheme from '../theme'
import { makeStyles, Grid, Typography, Paper, Hidden } from '@material-ui/core'
import Container from '@material-ui/core/Container'
import PhoneIphoneIcon from '@material-ui/icons/PhoneIphone'
import ComputerIcon from '@material-ui/icons/Computer'
import SectionHeader from './SectionHeader'
import { Transition } from 'react-transition-group'
import Slide from '@material-ui/core/Slide'
import DescriptionIcon from '@material-ui/icons/Description'
import GroupAddIcon from '@material-ui/icons/GroupAdd'
import MultilineChartIcon from '@material-ui/icons/MultilineChart'
import ShoppingCartIcon from '@material-ui/icons/ShoppingCart'

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: mainTheme.palette.defaultLight.main,
    paddingBottom: theme.spacing(5),
  },
  sectionTitle: {
    paddingBottom: theme.spacing(7),
    marginBottom: theme.spacing(6),
    background: '#0f0c29', //dark purple color
    color: mainTheme.palette.defaultLight.main,
    [theme.breakpoints.down('md')]: {
      marginBottom: theme.spacing(0),
      background: mainTheme.palette.defaultLight.main,
      color: mainTheme.palette.defaultDark.main,
    },
  },
  boxWrapper: {
    paddingTop: theme.spacing(6),
    [theme.breakpoints.down('md')]: {
      paddingTop: theme.spacing(0),
    },
  },
  boxes: {
    width: '400px',
    margin: theme.spacing(1),
    height: 'auto',
    padding: '.75rem 1.5rem',
    backgroundColor: '#ded5ef', // Light purple color
    // color: mainTheme.palette.defaultLight.main,
    color: '#818181',

    '&#box_0': {
      marginBottom: theme.spacing(-8),
      marginTop: theme.spacing(10),
      [theme.breakpoints.down('md')]: {
        margin: theme.spacing(1),
      },
    },
    '&#box_1': {
      [theme.breakpoints.down('md')]: {
        margin: theme.spacing(1),
      },
    },
    '&#box_2': {
      marginBottom: theme.spacing(-8),
      marginTop: theme.spacing(10),
      [theme.breakpoints.down('md')]: {
        margin: theme.spacing(1),
      },
    },
    '&#boxes_0': {
      marginTop: theme.spacing(-14),
      marginBottom: theme.spacing(16),
      [theme.breakpoints.down('md')]: {
        margin: theme.spacing(1),
      },
    },
    '&#boxes_1': {
      marginBottom: theme.spacing(5),
      marginTop: theme.spacing(-5),
      [theme.breakpoints.down('md')]: {
        margin: theme.spacing(1),
      },
    },
    '&#boxes_2': {
      marginTop: theme.spacing(-14),
      marginBottom: theme.spacing(16),
      [theme.breakpoints.down('md')]: {
        margin: theme.spacing(1),
      },
    },
  },
  icons: {
    width: '20%',
    height: 'auto',
    color: mainTheme.palette.defaultLight.main,
  },
  boxTitle: {
    paddingTop: theme.spacing(1),
    paddingBottom: theme.spacing(1),
  },
}))
const icons = [MultilineChartIcon, ComputerIcon, DescriptionIcon]
const icons2 = [PhoneIphoneIcon, ShoppingCartIcon, GroupAddIcon]
const serviceCardsOne = [
  {
    serviceTitle: 'SEO',
    serviceDetail:
      ' Lorem ipsum dolor, sit amet consectetur adipisicing elit.Placeat,facere, quasi animi rerum sed repellendus non ratione dignissimos consequuntur dolores?',
  },
  {
    serviceTitle: 'Website Development',
    serviceDetail:
      ' Lorem ipsum dolor, sit amet consectetur adipisicing elit.Placeat,facere, quasi animi rerum sed repellendus non ratione dignissimos consequuntur dolores?',
  },
  {
    serviceTitle: 'Content Writing',
    serviceDetail:
      ' Lorem ipsum dolor, sit amet consectetur adipisicing elit.Placeat,facere, quasi animi rerum sed repellendus non ratione dignissimos consequuntur dolores?',
  },
]
const serviceCardsTwo = [
  {
    serviceTitle: 'Mobile App Development',
    serviceDetail:
      ' Lorem ipsum dolor, sit amet consectetur adipisicing elit.Placeat,facere, quasi animi rerum sed repellendus non ratione dignissimos consequuntur dolores?',
  },
  {
    serviceTitle: 'eCommerce Development',
    serviceDetail:
      ' Lorem ipsum dolor, sit amet consectetur adipisicing elit.Placeat,facere, quasi animi rerum sed repellendus non ratione dignissimos consequuntur dolores?',
  },
  {
    serviceTitle: 'Maintance & Support',
    serviceDetail:
      ' Lorem ipsum dolor, sit amet consectetur adipisicing elit.Placeat,facere, quasi animi rerum sed repellendus non ratione dignissimos consequuntur dolores?',
  },
]
function HmServices() {
  const classes = useStyles()
  const [inProp, setInProp] = React.useState(false)

  const handleChange = () => {
    if (window.scrollY >= 750) {
      setInProp(true)
    } else {
      setInProp(false)
    }
  }
  window.addEventListener('scroll', handleChange)

  return (
    <div className={classes.root}>
      <Container maxWidth="xl">
        {/* Section title only shows on mobile */}
        <Hidden smUp>
          <SectionHeader
            title="Our Services"
            subTitle="Let us Help build the Future!"
          />
        </Hidden>

        <Grid
          container
          direction="row"
          justify="center"
          alignItems="center"
          className={classes.boxWrapper}
        >
          {serviceCardsOne.map((service, index) => {
            const Icon = icons[index]
            return (
              <Transition in={setInProp} timeout={1000} key={index}>
                {(state) => (
                  <Slide
                    direction="right"
                    in={inProp}
                    mountOnEnter
                    unmountOnExit
                    timeout={3000}
                  >
                    <Paper
                      elevation={10}
                      id={`box_${index}`}
                      className={classes.boxes}
                    >
                      <Icon className={classes.icons} />
                      <Typography variant="h5" className={classes.boxTitle}>
                        {service.serviceTitle}
                      </Typography>
                      <Typography paragraph>{service.serviceDetail}</Typography>
                    </Paper>
                  </Slide>
                )}
              </Transition>
            )
          })}
        </Grid>
        {/* Section Title shows on Ipads and up */}
        <Hidden xsDown>
          <Grid
            container
            direction="row"
            justify="center"
            className={classes.sectionTitle}
          >
            <SectionHeader
              title="Our Services"
              subTitle="Let us Help build the Future!"
            />
          </Grid>
        </Hidden>

        <Grid container direction="row" justify="center">
          {serviceCardsTwo.map((service, index) => {
            const Icon = icons2[index]
            return (
              <Transition in={setInProp} timeout={1000} key={index}>
                {(state) => (
                  <Slide
                    direction="left"
                    in={inProp}
                    mountOnEnter
                    unmountOnExit
                    timeout={3000}
                  >
                    <Paper
                      elevation={10}
                      id={`boxes_${index}`}
                      className={classes.boxes}
                    >
                      <Icon className={classes.icons} />
                      <Typography variant="h5" className={classes.boxTitle}>
                        {service.serviceTitle}
                      </Typography>
                      <Typography paragraph>{service.serviceDetail}</Typography>
                    </Paper>
                  </Slide>
                )}
              </Transition>
            )
          })}
        </Grid>
      </Container>
    </div>
  )
}

export default HmServices
