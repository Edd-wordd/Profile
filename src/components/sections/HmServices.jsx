import React from 'react'
import { Grid, Typography, Paper, Hidden, Container, Box } from '@mui/material'
import PhoneIphoneIcon from '@mui/icons-material/PhoneIphone'
import ComputerIcon from '@mui/icons-material/Computer'
import DescriptionIcon from '@mui/icons-material/Description'
import MultilineChartIcon from '@mui/icons-material/MultilineChart'
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart'
import GroupAddIcon from '@mui/icons-material/GroupAdd'
import serviceCardDetails from '../../data/hmServicesData'
import { SectionHeader } from '../index'
import { Slide } from '@mui/material'
import { useStyles } from '../styles/sections/HmServices.styles'
import { useTheme } from '@mui/system'

const icons = [
  MultilineChartIcon,
  ComputerIcon,
  DescriptionIcon,
  PhoneIphoneIcon,
  ShoppingCartIcon,
  GroupAddIcon,
]

function HmServices(props) {
  const theme = useTheme()
  const classes = useStyles(props)
  const [inProp, setInProp] = React.useState(false)
  const servicesSlide = (entries, observer) => {
    const [entry] = entries
    // guard clause
    if (!entry.isIntersecting) return
    if (entry.isIntersecting) {
      setInProp(true)
      observer.unobserve(entry.target)
    }
  }
  const servicesObserver = new IntersectionObserver(servicesSlide, {
    root: null,
    threshold: 0.2,
  })

  React.useEffect(() => {
    const target = document.querySelector('#services')
    servicesObserver.observe(target)
    return () => {
      servicesObserver.unobserve(target)
    }
  }, [])

  return (
    <div className={classes.root} id="services">
      <Container maxWidth="xl">
        <Box display={{ xs: 'block', sm: 'none', md: 'none', lg: 'none' }}>
          <SectionHeader title="Our Services" subTitle="Let us Help build the Future!" />
        </Box>
        <Grid
          container
          direction="row"
          justifyContent="center"
          alignItems="center"
          className={classes.cardWrapper}
        >
          <Box display={{ xs: 'none', sm: 'none', md: 'block', lg: 'block' }}>
            <Grid
              container
              direction="row"
              justifyContent="center"
              className={classes.sectionTitle}
            >
              <SectionHeader title="Our Services" subTitle="Let us Help build the Future!" />
            </Grid>
          </Box>
          {serviceCardDetails.map((service, index) => {
            const Icon = icons[index]
            const cardStyles = [
              {
                mb: 8,
                minHeight: '250px',
                [theme.breakpoints.down('md')]: {
                  m: 1,
                },
              },
              {
                mb: 17,
                minHeight: '250px',
                [theme.breakpoints.down('md')]: {
                  m: 1,
                },
              },
              {
                mb: 8,
                minHeight: '250px',
                [theme.breakpoints.down('md')]: {
                  m: 1,
                },
              },
              {
                mb: 10,
                minHeight: '250px',
                [theme.breakpoints.down('md')]: {
                  m: 1,
                },
              },
              {
                mb: 2,
                minHeight: '250px',
                [theme.breakpoints.down('md')]: {
                  m: 1,
                },
              },
              {
                mb: 10,
                minHeight: '250px',
                [theme.breakpoints.down('md')]: {
                  m: 1,
                },
              },
            ]
            return (
              <Slide
                key={service.id}
                direction={index >= 0 && index <= 2 ? 'right' : 'left'}
                in={inProp}
                mountOnEnter
                unmountOnExit
                timeout={3000}
              >
                <Paper
                  elevation={10}
                  id={`card_${index}`}
                  sx={{
                    position: 'relative',
                    width: '400px',
                    m: 1,
                    height: 'auto',
                    py: '0.75rem',
                    px: '1.5rem',
                    backgroundColor: theme.palette.primary_300.main,
                    color: theme.palette.charcoal.secondary,
                    ...cardStyles[index],
                  }}
                >
                  <Typography variant="h5" sx={{ display: 'flex', alignItems: 'center' }}>
                    <Icon className={classes.cardIcon} />
                    {service.serviceTitle}
                  </Typography>
                  <Typography paragraph>{service.serviceDescription}</Typography>
                </Paper>
              </Slide>
            )
          })}
        </Grid>
      </Container>
    </div>
  )
}

export default HmServices
