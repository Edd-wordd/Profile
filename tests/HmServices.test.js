import React from 'react'
import testComponent from './testComponent'
import HmServices from '../src/components/sections/HmServices'

testComponent(<HmServices />, 'HmServices')
