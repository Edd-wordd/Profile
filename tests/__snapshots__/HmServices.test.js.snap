// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HmServices should render HmServices 1`] = `"Component"`;
